foursight_core/routes.py

```diff
@@ -0,0 +1,231 @@
+    def history_route(environ, check):
+        """
+        Non-protected route
+        """
+        # get some query params
+        req_dict = app.current_request.to_dict()
+        query_params = req_dict.get('query_params')
+        start = int(query_params.get('start', '0')) if query_params else 0
+        limit = int(query_params.get('limit', '25')) if query_params else 25
+        domain, context = app.core.get_domain_and_context(req_dict)
+        return app.core.view_foursight_history(app.current_request, environ, check, start, limit,
+                                      app.core.check_authorization(req_dict, environ), domain, context)
+
+    @staticmethod
+    @app.route(ROUTE_PREFIX + 'checks/{environ}/{check}/{uuid}', methods=['GET'])
+    def get_check_with_uuid_route(environ, check, uuid):
+        """
+        Protected route
+        """
+        if app.core.check_authorization(app.current_request.to_dict(), environ):
+            return app.core.run_get_check(environ, check, uuid)
+        else:
+            return app.core.forbidden_response()
+
+    @staticmethod
+    @app.route(ROUTE_PREFIX + 'checks/{environ}/{check}', methods=['GET'])
+    def get_check_route(environ, check):
+        """
+        Protected route
+        """
+        if app.core.check_authorization(app.current_request.to_dict(), environ):
+            return app.core.run_get_check(environ, check, None)
+        else:
+            return app.core.forbidden_response()
+
+    @staticmethod
+    @app.route(ROUTE_PREFIX + 'checks/{environ}/{check}', methods=['PUT'])
+    def put_check_route(environ, check):
+        """
+        Take a PUT request. Body of the request should be a json object with keys
+        corresponding to the fields in CheckResult, namely:
+        title, status, description, brief_output, full_output, uuid.
+        If uuid is provided and a previous check is found, the default
+        behavior is to append brief_output and full_output.
+
+        Protected route
+        """
+        request = app.current_request
+        if app.core.check_authorization(request.to_dict(), environ):
+            put_data = request.json_body
+            return app.core.run_put_check(environ, check, put_data)
+        else:
+            return app.core.forbidden_response()
+
+    @staticmethod
+    @app.route(ROUTE_PREFIX + 'environments/{environ}', methods=['PUT'])
+    def put_environment(environ):
+        """
+        Take a PUT request that has a json payload with 'fourfront' (ff server)
+        and 'es' (es server).
+        Attempts to generate an new environment and runs all checks initially
+        if successful.
+
+        Protected route
+        """
+        request = app.current_request
+        if app.core.check_authorization(request.to_dict(), environ):
+            env_data = request.json_body
+            return app.core.run_put_environment(environ, env_data)
+        else:
+            return app.core.forbidden_response()
```

Comment:
I would disable this, as it is incompatible with EnvUtils at this time. We can create a ticket to make it compatible in the future.

Reply:
All set.